tools/structfield/structfield.go

```diff
@@ -23,60 +26,77 @@ import (
 }
 
-// JSONFieldNamePlugin is a custom linter plugin for golangci-lint.
-type JSONFieldNamePlugin struct {
-	allowedExceptions map[string]bool
+// StructFieldPlugin is a custom linter plugin for golangci-lint.
+type StructFieldPlugin struct {
+	allowedTagNameExceptions map[string]bool
+	allowedTagTypeExceptions map[string]bool
 }
 
-// Settings is the configuration for the jsonfieldname linter.
+// Settings is the configuration for the structfield linter.
 type Settings struct {
-	AllowedExceptions []string `json:"allowed-exceptions" yaml:"allowed-exceptions"`
+	AllowedTagNameExceptions []string `json:"allowed-tag-name-exceptions" yaml:"allowed-tag-name-exceptions"`
+	AllowedTagTypeExceptions []string `json:"allowed-tag-type-exceptions" yaml:"allowed-tag-type-exceptions"`
 }
 
 // New returns an analysis.Analyzer to use with golangci-lint.
-// It parses the "allowed-exceptions" section to determine which warnings to skip.
 func New(cfg any) (register.LinterPlugin, error) {
-	allowedExceptions := map[string]bool{}
+	allowedTagNameExceptions := map[string]bool{}
+	allowedTagTypeExceptions := map[string]bool{}
 
 	if cfg != nil {
 		if settingsMap, ok := cfg.(map[string]any); ok {
-			if exceptionsRaw, ok := settingsMap["allowed-exceptions"]; ok {
+			if exceptionsRaw, ok := settingsMap["allowed-tag-name-exceptions"]; ok {
 				if exceptionsList, ok := exceptionsRaw.([]any); ok {
 					for _, item := range exceptionsList {
 						if exception, ok := item.(string); ok {
-							allowedExceptions[exception] = true
+							allowedTagNameExceptions[exception] = true
+						}
+					}
+				}
+			}
+
+			if exceptionsRaw, ok := settingsMap["allowed-tag-type-exceptions"]; ok {
+				if exceptionsList, ok := exceptionsRaw.([]any); ok {
+					for _, item := range exceptionsList {
+						if exception, ok := item.(string); ok {
+							allowedTagTypeExceptions[exception] = true
 						}
 					}
 				}
 			}
 		}
 	}
 
-	return &JSONFieldNamePlugin{allowedExceptions: allowedExceptions}, nil
+	return &StructFieldPlugin{
+		allowedTagNameExceptions: allowedTagNameExceptions,
+		allowedTagTypeExceptions: allowedTagTypeExceptions,
+	}, nil
 }
 
-// BuildAnalyzers builds the analyzers for the JSONFieldNamePlugin.
-func (f *JSONFieldNamePlugin) BuildAnalyzers() ([]*analysis.Analyzer, error) {
+// BuildAnalyzers builds the analyzers for the StructFieldPlugin.
+func (f *StructFieldPlugin) BuildAnalyzers() ([]*analysis.Analyzer, error) {
 	return []*analysis.Analyzer{
 		{
-			Name: "jsonfieldname",
-			Doc:  "Reports mismatches between Go field and JSON tag names. Note that the JSON tag name is the source-of-truth and the Go field name needs to match it.",
+			Name: "structfield",
+			Doc: `Reports mismatches between Go field and JSON or URL tag names and types.
+Note that the JSON or URL tag name is the source-of-truth and the Go field name needs to match it.
+If the tag contains "omitempty", then the Go field must be a pointer type except slices and maps.`,
```

Comment:
An alternative to this wording might be "must be a reference type."... and then you don't need to make an exception since slices and maps are already reference types.

Your call... I'm fine either way.